src/Driver/FetchUtils.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Driver;
+
+use Traversable;
+
+/**
+ * @internal
+ */
+final class FetchUtils
+{
+    /**
+     * @return mixed|false
+     *
+     * @throws DriverException
+     */
+    public static function fetchOneFromNumeric(ResultStatement $stmt)
+    {
+        $row = $stmt->fetchNumeric();
+
+        if ($row === false) {
+            return false;
+        }
+
+        return $row[0];
+    }
+
+    /**
+     * @return array<int,array<int,mixed>>
+     *
+     * @throws DriverException
+     */
+    public static function fetchAllNumericByOne(ResultStatement $stmt) : array
```

Comment:
Why the `ByOne` suffix here and below? I don't understand what it means.

Reply:
These methods implement `fetchAll*()` statement methods by fetching each row individually (one by one). Not sure how to put it clear and compact at the same time.

Reply:
It's an implementation detail, isn't it? Why does it need to be reflected in the name of the method? And if it really does, since all methods in that class seem to have that same behavior, let's maybe reflect it in the class name? `OneByOneFetcher`, `IteratingFetcher`, something like that?

Reply:
> It's an implementation detail, isn't it? Why does it need to be reflected in the name of the method?

That's a good point. In the beginning, I thought we might have different implementations of the same fetch mode for different cases (e.g. “fetch one value from a numeric array” vs “fetch one value from an associative array”) but it didn't happen. We can remove the suffixes and leave just the methods matching the statement method names.

Would that be okay?

Reply:
Sounds perfect!